### ISSUE TITLE
Resteasy-reactive routing does not work when paths are shared across classes.

### ISSUE THREAD

**Issue by cogman** [quarkusio/quarkus#18542]
## Describe the bug
Having 1 route handled by 2 classes doesn't work (It does in plain resteasy).  For example, 

- GET /hello/{id} in class A
- GET /hello/{id}/foo in class B

Attempting class A's route results in a 404.

### Expected behavior
Routing to Class A's route

### Actual behavior
404

## To Reproduce

[route-bug.zip](https://github.com/quarkusio/quarkus/files/6787022/route-bug.zip)

1. Start quarkus
2. Visit /hello/123 observe 404
3. Visit /hello/123/bar receive nice message

### Screenshots
![Resource Not Found](https://user-images.githubusercontent.com/4906592/124984809-ea40db80-dff6-11eb-9993-a35691ee9a75.png)
![hello](https://user-images.githubusercontent.com/4906592/124984799-e7de8180-dff6-11eb-89c9-02cbadfe5b7a.png)

## Environment (please complete the following information):
windows and linux

### Output of `uname -a` or `ver`
Linux CWAN-G3NH0Z2 4.19.128-microsoft-standard #1 SMP Tue Jun 23 12:58:10 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`
openjdk 11.0.11 2021-04-20
OpenJDK Runtime Environment GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05)
OpenJDK 64-Bit Server VM GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05, mixed mode, sharing)

### Quarkus version or git rev
2.0.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
Doesn't matter, reproduces in both.
But, maven 3.8.1 and gradle 7.1

**Comment by cogman** [comment]
Note, this structure makes a lot more sense in our product.  We have GET/POST/PUT/DELETE verbs on the naked path and further descriptions of the resource in the other class.

**Comment by geoand** [comment]
Does the same thing work with `quarkus-resteasy`?

**Comment by stuartwdouglas** [comment]
This should work according to the spec, I am looking into it.

**Comment by cogman** [comment]
@geoand Yes, this works currently with `quarkus-resteasy` I ran into this transitioning a project over to the reactive rest easy to get kotlin coroutine support.

For anyone that runs into this issue, A work around is to make new classes with the path parameter on the class instead of as a method parameter.

For example, this works

```java
@Path("hello/{id}")
public class FooResource {
   @GET
   public void hello(@PathParam("id") String id) {}
}
```

This does not

```java
@Path("hello")
public class FooResource {
   @GET
   @Path("{id}")
   public void hello(@PathParam("id") String id) {}
}
```

**Comment by cogman** [comment]
@stuartwdouglas has there been any progress in looking into this?  Is there a way I can help?

**Comment by stuartwdouglas** [comment]
Sorry, this fell of my radar. Should be fixed with the linked PR.

**Comment by lucashsilva** [comment]
@gsmet @stuartwdouglas In which version is this fix available? I'm running 2.9.2.Final and it does not seem to work.
